Av old hyrax

jwellnit opened the pull request:
Add audio visual work type and associated back-end support

jwellnit committed:
empty video model

jwellnit committed:
stripped AV, no oembed

jwellnit committed:
back to basics

jwellnit committed:
getting presigned url

jwellnit committed:
new player (possibly unnecessary) and aws changes

jwellnit committed:
working presigned url

jwellnit committed:
temporarily use bulkrax bucket

jwellnit committed:
add premade derivative metadata

jwellnit committed:
derivatives, partially working

jwellnit committed:
derivatives working

jwellnit committed:
Reconnected Front end, still some local issues

jwellnit committed:
temporarily use bulkrax bucket again

jwellnit committed:
linter fix 1

jwellnit committed:
merged audio and video derivatives for code climate

jwellnit committed:
linter fix 2

jwellnit committed:
code climate fix 1

jwellnit committed:
Code Climate fix 2

jwellnit committed:
Code Climate fix 3

jwellnit committed:
Code Climate fix 4

rococodogs committed:
use localhost hostname for s3 urls in development

jwellnit committed:
several small fixes

-av objects require files
-testing with specified resolution for derivatives
-description of av work type

jwellnit committed:
Linter fix 3

jwellnit committed:
not working, save

jwellnit committed:
bitrate

jwellnit committed:
redo size

jwellnit committed:
syntax

jwellnit committed:
last ditch effort

jwellnit committed:
partial upgrade for hydra, derivative quality can be modified now

jwellnit committed:
generate multiple derivatives

jwellnit committed:
multi video view

jwellnit committed:
multi value premade derivatives

jwellnit committed:
variable derivative resolution

jwellnit committed:
fixed viewer sizing and loading, added stored_derivatives field

jwellnit committed:
initial pass on stored derivatives field

jwellnit committed:
add to stored derivatives

jwellnit committed:
small adjustments and fixes

jwellnit committed:
broken

jwellnit committed:
variable sources with premade derivatives

jwellnit committed:
derivative cleanup works

jwellnit committed:
naming conventions and quality labels

jwellnit committed:
broken, save for reference

jwellnit committed:
currently only running on initial visit

still doesn't apply options

jwellnit committed:
service file split and video view fix

jwellnit committed:
linter and cleanup fixes

jwellnit committed:
linter fix 2

jwellnit committed:
linter fix 3

jwellnit committed:
first pass testing on actor, controller, form, and indexer

jwellnit committed:
small fixes

jwellnit committed:
first pass working playlist

jwellnit committed:
table view for playlist

jwellnit committed:
not working

save for ref

jwellnit committed:
working names

jwellnit committed:
handles multi file sets

jwellnit committed:
lint fix

jwellnit committed:
lint 2

jwellnit committed:
testing 1

jwellnit committed:
testing 2

jwellnit committed:
removed failing tests temporarily

jwellnit committed:
temporarily remove all new tests

jwellnit committed:
model nested properties

jwellnit committed:
actor spec

jwellnit committed:
controller spec

jwellnit committed:
added av factory

jwellnit committed:
av form spec

jwellnit committed:
add back resource type

jwellnit committed:
form spec fixes

jwellnit committed:
av indexer spec

jwellnit committed:
index stored_derivatives

jwellnit committed:
Presenter tests

jwellnit committed:
model tests

jwellnit committed:
lint

jwellnit committed:
remove model tests

jwellnit committed:
adding the model tests back?

jwellnit committed:
av helper test 1

jwellnit committed:
av helper test 1.1

jwellnit committed:
av helper test 1.2

jwellnit committed:
av helper test 1.3

jwellnit committed:
av helper test 1.4

jwellnit committed:
features test

jwellnit committed:
features test 1.1

jwellnit committed:
av helper test 2

jwellnit committed:
av helper test 2.1

jwellnit committed:
av helper test 2.2

jwellnit committed:
av helper 2.3

jwellnit committed:
av helper test 4

jwellnit committed:
base files for derivative service specs

jwellnit committed:
valid test for base derivatives

jwellnit committed:
quick fix

jwellnit committed:
inherited tests

jwellnit committed:
actually inherited tests

jwellnit committed:
cleanup tests

jwellnit committed:
lint

jwellnit committed:
quick fix

jwellnit committed:
derivative url test

jwellnit committed:
s3 tests

jwellnit committed:
lint 1

jwellnit committed:
lint 2

jwellnit committed:
lint 3

jwellnit committed:
lint 4

jwellnit committed:
video inherited tests

jwellnit committed:
audio inherited tests

jwellnit committed:
path and url tests

jwellnit committed:
resolution tests

jwellnit committed:
lint

jwellnit committed:
check premade tests

jwellnit committed:
lint

jwellnit committed:
rename derivatives tests

jwellnit committed:
create derivative test

jwellnit committed:
lint 1

jwellnit committed:
lint 3

jwellnit committed:
lint 4

jwellnit committed:
lock file

jwellnit committed:
video processor tests

jwellnit committed:
lint

jwellnit committed:
rename premade derivative test improvements

jwellnit committed:
lint

jwellnit committed:
original name test improvements

jwellnit committed:
lint

jwellnit committed:
s3 url tests

jwellnit committed:
lint

jwellnit committed:
s3 url tests 2

jwellnit committed:
lint

jwellnit committed:
stub env

jwellnit committed:
s3 url tests take 3

rococodogs commented on app/helpers/audio_visual_helper.rb:
```diff
@@ -0,0 +1,47 @@
+      return ""
+    end
+    client_opts = {}
+    client_opts = { endpoint: ENV['AWS_ENDPOINT_URL']&.sub('minio', 'localhost') } if Rails.env.development?
```
try pulling out the `&` after `ENV['AWS_ENDPOINT_URL']` to fix coverage? since you're already testing to see if it's empty above, you don't need to assume it might not be there at this point

jwellnit committed:
&

jwellnit committed:
possible fix for remove block test coverage

jwellnit committed:
audio create deriv file remove tests

jwellnit committed:
lint

jwellnit committed:
the other remove file tests

jwellnit committed:
lint

jwellnit committed:
lint 2

jwellnit committed:
video processor audio attribute tests

jwellnit committed:
more video processor tests

jwellnit committed:
remove custom bitrate with default video attr

not used and would require maintaining further customizations (the config file)

jwellnit committed:
partial fix, still broken

jwellnit committed:
undo helper change

still broken

jwellnit committed:
possible fix

jwellnit committed:
lint

jwellnit committed:
redo s3 url tests

jwellnit committed:
more test fixes

jwellnit committed:
lint

jwellnit committed:
endpoint variable only expected in develop

jwellnit committed:
lint

jwellnit committed:
return with error messages

rococodogs committed:
allow file_sets to store transcript along-side file

jwellnit committed:
metadata first pass

jwellnit committed:
remove format

jwellnit committed:
labels and language fix

rococodogs committed:
lmao rubo

rococodogs committed:
add comments

rococodogs committed:
add spec

rococodogs committed:
rubo

rococodogs committed:
add transcript attachment service

rococodogs committed:
smdh

jwellnit committed:
metadata tests first pass

jwellnit committed:
lint

jwellnit committed:
test and view fixes

jwellnit committed:
location stub

jwellnit committed:
better location stub

jwellnit committed:
lint

jwellnit committed:
lint 2

jwellnit committed:
lint 3

jwellnit committed:
please

jwellnit committed:
test multi auth for av location

jwellnit committed:
quick fix

jwellnit committed:
stub

jwellnit committed:
lint

jwellnit committed:
date encompassed test av

jwellnit committed:
provenance visible

jwellnit committed:
controller nocov

jwellnit committed:
captions first pass

jwellnit committed:
remove 10 sec limiter

jwellnit committed:
error correcting, better string splicing, fileset issue fix

jwellnit committed:
working transcripts

jwellnit committed:
transcript name flag

jwellnit committed:
download transcript

jwellnit committed:
lint

jwellnit committed:
spec fixes

jwellnit committed:
new tests first pass

jwellnit committed:
lint

jwellnit committed:
s3 spec fix

jwellnit committed:
premade derivatives transcript fix

jwellnit committed:
file set edit now uses thumbnail

jwellnit committed:
caption badge

jwellnit committed:
transcript downloads

jwellnit committed:
stored and premade derivatives refactor

jwellnit committed:
lint

jwellnit committed:
spec pass 1

jwellnit committed:
oops

jwellnit committed:
spec pass 2

jwellnit committed:
spec pass 3

jwellnit committed:
stupid mistake

jwellnit committed:
typo

jwellnit committed:
end

jwellnit committed:
wrong bucket

jwellnit committed:
quick test

jwellnit committed:
lint

jwellnit committed:
have_recieved

jwellnit committed:
audio visual service fixes

jwellnit committed:
typo

jwellnit committed:
work exporter tests 1

jwellnit committed:
lint

jwellnit committed:
fix

jwellnit committed:
work exporter tests 2

jwellnit committed:
ends

jwellnit committed:
lint

jwellnit committed:
files

jwellnit committed:
potential fix

jwellnit committed:
array

jwellnit committed:
lock

jwellnit committed:
derivative and caption naming

jwellnit committed:
potential audio derivative test fix

jwellnit committed:
small  fix

jwellnit committed:
video derivative fixes

jwellnit committed:
transcript test fix

jwellnit committed:
second try

jwellnit committed:
debug logging

jwellnit committed:
warn edit

jwellnit committed:
one more time

jwellnit committed:
temporarily disable a test because oh my god

jwellnit committed:
try logger

jwellnit committed:
sidekiq logger

jwellnit committed:
eyeroll

jwellnit committed:
alt format

jwellnit committed:
try hyrax logger

jwellnit committed:
Hyrax log change and file touch

rococodogs committed:
make sure s3 download folder exists before dl

jwellnit committed:
test changes 1

jwellnit committed:
lint

jwellnit committed:
tmp

rococodogs committed:
use Rails.root.join for tmp path

jwellnit committed:
update

rococodogs committed:
rubify some places + add comments (#1160)

* rubify some places + add comments

* d'oh rubo

* spec fixes

* more spec fixes

---------

Co-authored-by: Jennifer Wellnitz <jennifer.wellnitz22@gmail.com>
